File: app/src/main/java/com/aa/mynotes/DBOpenHelper.java
package com.aa.mynotes;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;


public class DBOpenHelper extends SQLiteOpenHelper {
    //Constants for db name and version
    private static final String DATABASE_NAME = "notes.db";
    private static final int DATABASE_VERSION = 1;

    //Constants for identifying table and columns
    public static final String TABLE_NOTES = "notes";
    public static final String NOTE_ID = "_id";
    public static final String NOTE_TEXT = "noteText";
    public static final String NOTE_LAST_CHANGED = "noteLastChanged";

    public static final String[] ALL_COLUMNS = {NOTE_ID, NOTE_TEXT, NOTE_LAST_CHANGED};

    //SQL to create table
    private static final String TABLE_CREATE =
            "CREATE TABLE " + TABLE_NOTES + " (" +
                    NOTE_ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                    NOTE_TEXT + " TEXT, " +
                    NOTE_LAST_CHANGED + " TEXT default (datetime(CURRENT_TIMESTAMP, 'localtime'))" +
                    ")";

    public DBOpenHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    /**
     * The onCreate method is called the first time this class is instantiated and it result in
     * creating the database structure id it doesn't already exist.
     * @param db
     */
    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(TABLE_CREATE);
    }

    /**
     * The onUpgrade method is called when you've changed the database version
     * and the user opens the app for the first time after that happens.
     * You can put code in the onUpgrade method that can rebuild the database with the new structure.
     * @param db
     * @param oldVersion
     * @param newVersion
     */
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NOTES);
        onCreate(db);
    }
}

File: app/src/main/java/com/aa/mynotes/data/DBOpenHelper.kt
package com.aa.mynotes.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBOpenHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(TABLE_CREATE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS $TABLE_NOTES")
        onCreate(db)
    }

    companion object {
        private const val DATABASE_NAME = "notes.db"
        private const val DATABASE_VERSION = 1

        const val TABLE_NOTES = "notes"
        const val NOTE_ID = "_id"
        const val NOTE_TEXT = "noteText"
        const val NOTE_LAST_CHANGED = "noteLastChanged"

        @JvmField
        val ALL_COLUMNS = arrayOf(NOTE_ID, NOTE_TEXT, NOTE_LAST_CHANGED)

        private const val TABLE_CREATE =
            "CREATE TABLE $TABLE_NOTES (" +
                    "$NOTE_ID INTEGER PRIMARY KEY AUTOINCREMENT, " +
                    "$NOTE_TEXT TEXT, " +
                    "$NOTE_LAST_CHANGED TEXT default (datetime(CURRENT_TIMESTAMP, 'localtime'))" +
                    ")"
    }
}
